hermes-api/src/main/java/pl/allegro/tech/hermes/api/TraceInfo.java

```diff
@@ -25,12 +57,64 @@ public boolean equals(Object o) {
 
         TraceInfo traceInfo = (TraceInfo) o;
 
-        return !(traceId != null ? !traceId.equals(traceInfo.traceId) : traceInfo.traceId != null);
+        if (traceId != null ? !traceId.equals(traceInfo.traceId) : traceInfo.traceId != null) return false;
```

Comment:
Those statements are not easy to read - too many negations. Maybe we could use here java.util.Objects.equals()